### PR TITLE
Add doctor dbconsistency fix to delete repos with no owner (#27290)

### PR DESCRIPTION
**Pull request by 6543**
Backport #27290

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

File: modules/doctor/repository.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	e := db.GetEngine(ctx)
+	var deleted int64
+	adminUser := &user_model.User{IsAdmin: true}
+
+	for {
+		var ids []int64
+		if err := e.Table("`repository`").
+			Join("LEFT", "`user`", "repository.owner_id=user.id").
+			Where(builder.IsNull{"`user`.id"}).
+			Select("`repository`.id").Limit(batchSize).Find(&ids); err != nil {
```
Comment (JakobDev) [inline modules/doctor/repository.go:44]: I think you forget a where for the ID. It looks like you just get the first n repositories of the batch size and ignore the rest.

Comment (KazzmanK) [reply modules/doctor/repository.go:44]: Isn't absence of user.id for _any_ repository is what you are looking for ?
`Where(builder.IsNull{"`user`.id"}).`

Comment (JakobDev) [reply modules/doctor/repository.go:44]: I mean the Repository ID

Comment (6543) [reply modules/doctor/repository.go:44]: @JakobDev I utilize the left join to filter out the broken repos

and yes I limit it as it is run in a for loop ... so its intendet

Comment (6543) [reply modules/doctor/repository.go:44]: I dont get the issue ... it's working as intendet.
could you provide a code diff or better explanation what issue you did discover please :)

Comment (JakobDev) [reply modules/doctor/repository.go:44]: Was a mistake by me. Sorry. Normally you need to run something like this to get all repos:
```sql
SELECT * FROM repository WHERE id > 0 ORDER BY id LIMIT 100
SELECT * FROM repository WHERE id > 100 ORDER BY id LIMIT 100
SELECT * FROM repository WHERE id > 200 ORDER BY id LIMIT 100
```
but in this case everything what you get with this Query is deleted, so you catchall repos without duplicates.

**Review by kerwin612 — APPROVED** [review]

**Comment by 6543** [comment]
:mage:

**Pull request merged**